torch/ao/quantization/fx/_model_report/_detector.py

```diff
@@ -134,9 +119,151 @@ def _detect_per_channel_helper(model: nn.Module):
+    """
+    determines whether dynamic or static quantization is more appropriate for a given module
+
+    Stationary distribution of data are strictly above tolerance level for the comparison statistic:
+
+        S = average_batch_activation_range/epoch_activation_range
+
+    Nonstationary distributions are below the tolerance level for this metric
+
+    This will then generate suggestions for dynamic vs static quantization focused around Linear
+
+    Args:
+        model: The prepared and calibrated GraphModule with inserted ModelReportObservers around layers of interest
+
+    """
+
+    # store modules dynamic vs static information
+    module_dynamic_static_info = {}
+
+    # loop through all submodules included nested ones
+    for name, module in model.named_modules():
+        # if module has the ModelReportObserver attached to it
+        if hasattr(module, DEFAULT_PRE_OBSERVER_NAME) and hasattr(module, DEFAULT_POST_OBSERVER_NAME):
+            # get pre and post observers for the module
+            pre_obs = getattr(module, DEFAULT_PRE_OBSERVER_NAME)
+            post_obs = getattr(module, DEFAULT_POST_OBSERVER_NAME)
+
+            # get the statistics for each module
+            pre_stat = pre_obs.get_batch_to_epoch_ratio()
+            post_stat = post_obs.get_batch_to_epoch_ratio()
+
+            # record module, pre and post stat, and whether to do dynamic or static based off it
+            dynamic_recommended = False
+
+            # string for pre and post observer classification
+            pre_obs_dist_classification = ""
+            post_obs_dist_classification = ""
+
+            if pre_stat > tolerance and post_stat > tolerance:
+                dynamic_recommended = False  # static is best if both stationary
+                pre_obs_dist_classification = DEFAULT_STATIONARY
+                post_obs_dist_classification = DEFAULT_STATIONARY
+            elif pre_stat <= tolerance and post_stat > tolerance:
+                dynamic_recommended = False  # static best if input non-stationary, output stationary
+                pre_obs_dist_classification = DEFAULT_NON_STATIONARY
+                post_obs_dist_classification = DEFAULT_STATIONARY
+            elif pre_stat <= tolerance and post_stat <= tolerance:
+                dynamic_recommended = True  # dynamic best if input, output non-stationary
+                pre_obs_dist_classification = DEFAULT_NON_STATIONARY
+                post_obs_dist_classification = DEFAULT_NON_STATIONARY
+            elif pre_stat > tolerance and post_stat <= tolerance:
+                dynamic_recommended = True  # dynamic best if input stationary, output non-stationary
+                pre_obs_dist_classification = DEFAULT_STATIONARY
+                post_obs_dist_classification = DEFAULT_NON_STATIONARY
+            else:
+                raise Exception("Should always take one of above branches")
```

Comment:
we'll never reach here? if so we can change the last `elif` to else and comment with conditions